comp/forwarder/defaultforwarder/domain_forwarder_test.go

```diff
@@ -121,15 +121,12 @@ func TestDomainForwarderSendHTTPTransactions(t *testing.T) {
 }
 
 func TestDomainForwarderHAPreFailover(t *testing.T) {
+	mockConfig := mock.New(t)
```

Comment:
After looking at more of the tests, I think we need some easy way to build a config mock by parsing yaml. Especially for tests that use lists in the config; it's awkward to use `Set` to build complex types in a list. For example see https://github.com/DataDog/datadog-agent/blob/fe75b815c2f135f0d2ea85d7a57a8fc8cbf56bd9/pkg/config/utils/metadata_providers.go#L22 It's much easier to test this by defining a list of MetadataProviders in yaml.

But rather than have a proliferation of tests that parse yaml to build a config, we should have only 1 way to do it, preferably in the `pkg/config/mock/` package, and all the tests throughout the codebase should use that.

Reply:
I added a method in the mock package.